src/main/kotlin/com/ampnet/blockchainapiservice/util/Resolvable.kt

```diff
@@ -0,0 +1,7 @@
+package com.ampnet.blockchainapiservice.util
+
+import com.ampnet.blockchainapiservice.exception.IncompleteRequestException
+
+data class Resolvable<T>(val alternative: T?, val message: String) {
+    fun resolve(value: T?): T = value ?: alternative ?: throw IncompleteRequestException(message)
+}
```

Comment:
Got it iteratively, first I extracted logic into a method in `StoreErc20BalanceRequestParams` which looked something like this: `fun <T> get(value: T?, alternative: R?, message: String): T = ...` but then I had to do something like `get(params.chainId, clientInfo.chainId, "Missing chainId")` which is in itself fine but I relized I would have to duplicate it in `StoreSendErc20RequestParams`... So I decided to move that logic into `ClientInfo`. And there it was kind of ugly to have to pass a reference to the field, something like this: `fun <T> resolve(value: T, alternative: ClientInfo.() -> T, message: String): T = ...` and you would call it like: `clientInfo.resolve(chainId, ClientInfo::chainId, "Missing chainId"`. I thought it would be nice to do something like `clientInfo.<field>.resolve(...)` so that's how I got to this.